applications/nrf5340_audio/src/modules/hw_codec.c

```diff
@@ -26,10 +28,97 @@ LOG_MODULE_REGISTER(hw_codec, CONFIG_MODULE_HW_CODEC_LOG_LEVEL);
 #define VOLUME_ADJUST_STEP_DB 3
 #define BASE_10		      10
 
+ZBUS_SUBSCRIBER_DEFINE(volume_evt_sub, CONFIG_VOLUME_MSG_SUB_QUEUE_SIZE);
+
 static uint32_t prev_volume_reg_val = OUT_VOLUME_DEFAULT;
 
 static cs47l63_t cs47l63_driver;
 
+static k_tid_t volume_msg_sub_thread_id;
+static struct k_thread volume_msg_sub_thread_data;
+
+K_THREAD_STACK_DEFINE(volume_msg_sub_thread_stack, CONFIG_VOLUME_MSG_SUB_STACK_SIZE);
+
+/**
+ * @brief  Convert zbus volume to actual volume setting for HW codec.
+ *
+ *         This range for zbus volume is from 0 to 255 and the
+ *         range for HW codec volume is from 0 to 128, this function
+ *         converting the zbus volume to HW codec volume setting.
+ */
+static uint16_t zbus_vol_conversion(uint8_t volume)
+{
+	return (((uint16_t)volume + 1) / 2);
+}
+
+/* Handle volume events from zbus */
+static void volume_msg_sub_thread(void)
+{
+	int ret;
+
+	const struct zbus_channel *chan;
+
+	while (1) {
+		ret = zbus_sub_wait(&volume_evt_sub, &chan, K_FOREVER);
+		ERR_CHK(ret);
+
+		struct volume_msg msg;
+
+		ret = zbus_chan_read(chan, &msg, K_MSEC(100));
```

Comment:
Should we have a globally defined timeout here?

Reply:
Sure, can be defined in nrf5340_audio_common.h and then moved to the "z-bus file" once that is created